src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/HearingCentreFinder.java

```diff
@@ -46,10 +50,28 @@ public HearingCentre find(
                     .findFirst();
 
             if (hearingCentre.isPresent()) {
-                return hearingCentre.get();
+
+                final String hearingCentreActivationDate =
+                    hearingCentreActivationDates.get(hearingCentre.get());
+
+                if (hearingCentreIsActive(hearingCentreActivationDate)) {
+                    return hearingCentre.get();
+                }
             }
         }
 
         return defaultHearingCentre;
     }
+
+    public boolean hearingCentreIsActive(String hearingCentreActivationDate) {
```

Comment:
Do we care about the time here?  Does the activation for instance intended to happen after 00:00 on the configured date or the next day?